GP/kernel_modules/random_walk.py

```diff
@@ -0,0 +1,102 @@
+# Author: Henry Moss & Ryan-Rhys Griffiths
```

Comment:
Having a `kernel_modules` directory is probably a good idea.

Reply:
I'm happy for it to have another (less clumsy) name but yeah I think it was the only way for me to implement the kernel without contaminating pre-existing code